x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/NerConfig.java

```diff
@@ -0,0 +1,141 @@
+        return LENIENT_PARSER.apply(parser, null);
+    }
+
+    private static final ConstructingObjectParser<NerConfig, Void> STRICT_PARSER = createParser(false);
+    private static final ConstructingObjectParser<NerConfig, Void> LENIENT_PARSER = createParser(true);
+
+    @SuppressWarnings({ "unchecked"})
+    private static ConstructingObjectParser<NerConfig, Void> createParser(boolean ignoreUnknownFields) {
+        ConstructingObjectParser<NerConfig, Void> parser = new ConstructingObjectParser<>(NAME, ignoreUnknownFields,
+            a -> new NerConfig((VocabularyConfig) a[0], (TokenizationParams) a[1], (List<String>) a[2]));
+        parser.declareObject(ConstructingObjectParser.constructorArg(), VocabularyConfig.createParser(ignoreUnknownFields), VOCABULARY);
+        parser.declareObject(ConstructingObjectParser.optionalConstructorArg(), TokenizationParams.createParser(ignoreUnknownFields),
+            TOKENIZATION_PARAMS);
+        parser.declareStringArray(ConstructingObjectParser.optionalConstructorArg(), CLASSIFICATION_LABELS);
+        return parser;
+    }
+
+    private final VocabularyConfig vocabularyConfig;
+    private final TokenizationParams tokenizationParams;
+    private final List<String> classificationLabels;
+
+    public NerConfig(VocabularyConfig vocabularyConfig,
+                     @Nullable TokenizationParams tokenizationParams,
+                     @Nullable List<String> classificationLabels) {
+        this.vocabularyConfig = ExceptionsHelper.requireNonNull(vocabularyConfig, VOCABULARY);
+        this.tokenizationParams = tokenizationParams == null ? TokenizationParams.createDefault() : tokenizationParams;
+        this.classificationLabels = classificationLabels;
+    }
+
+    public NerConfig(StreamInput in) throws IOException {
+        vocabularyConfig = new VocabularyConfig(in);
+        tokenizationParams = new TokenizationParams(in);
+        classificationLabels = in.readOptionalStringList();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(VOCABULARY.getPreferredName(), vocabularyConfig);
+        builder.field(TOKENIZATION_PARAMS.getPreferredName(), tokenizationParams);
+        if (classificationLabels != null) {
+            builder.field(CLASSIFICATION_LABELS.getPreferredName(), classificationLabels);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        vocabularyConfig.writeTo(out);
+        tokenizationParams.writeTo(out);
+        out.writeOptionalStringCollection(classificationLabels);
+    }
+
+    @Override
+    public boolean isTargetTypeSupported(TargetType targetType) {
+        return false;
+    }
```

Comment:
This and all other of these checks, I am thinking we need to add a new `TargetType` for `NLP` or something. We shouldn't allow these types of configs for the boosted tree models. 

I am not 100% sure the best way here.

Reply:
Agreed. I'm planning to address this in a follow-up PR where we have validation on the model_type -> inference_config combinations.